HARK/distribution.py

```diff
@@ -1249,6 +1298,16 @@ def calc_expectation(dstn,func=lambda x : x,*args):
 
     return f_exp
 
+# "calc_expectation" is not a good name for something that
+# requires an array as an argument, returns an array, and
+# expects a peculiarly shaped distribution object as its first
+# argument.
+# TODO: 20210618: we REALLY need to improve our tools for calculating
+# expectations, both to make them better documented and more flexible.
+
+
+calc_expectation_of_array = calc_expectation
+
```

Comment:
Again, this is not the appropriate place to enjoin a debate about the design of this function, which I agree could use more work.

I also disagree with the opinions in the comment.

The current form of this function does not require an array as an argument, for starters.